src/Microsoft.PowerShell.Commands.Utility/commands/utility/NewTemporaryFileCommand.cs

```diff
@@ -0,0 +1,46 @@
+
+using System;
+using System.IO;
+using System.Management.Automation;
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// The implementation of the "New-TemporaryFile" cmdlet
+    /// </summary>    
+    [Cmdlet(VerbsCommon.New, "TemporaryFile", SupportsShouldProcess = true, HelpUri = "https://go.microsoft.com/fwlink/?LinkId=526726")]
+    [OutputType(typeof(System.IO.FileInfo))]
+    public class NewTemporaryFileCommand : Cmdlet
+    {      
+        /// <summary>
+        /// returns a TemporaryFile
+        /// </summary>
+        protected override void BeginProcessing()
+        {
+            string filePath = null;
+            if (ShouldProcess(System.Environment.GetEnvironmentVariable("TEMP")))
+            {
+                try
+                {
+                    filePath = Path.GetTempFileName();                
+                }
+                catch (Exception e)
+                {
+                    WriteError(
+                        new ErrorRecord(
+                            e,
+                            "NewTemporaryFileWriteError",
+                            ErrorCategory.WriteError,
+                            System.Environment.GetEnvironmentVariable("TEMP")));
+                    return;
+                }
+                //filePath may be null, only writeobject if there is a value
```

Comment:
We usually use a space after `//`.
And the comment is obvious and likely unnecessary.